github-management/default-branch-migration.md

```diff
@@ -0,0 +1,110 @@
+# Default Branch Migration
+
+This document outlines steps needed to migrate the default branch
+of your repo from `master` to `main`.
+
+Note: This document is currently a work in progress.
+
+If you have questions about the process, reach out to the GitHub Management Team
+on the _#github-management_ channel on slack or open an issue in the [kubernetes/org] repo.
+
+## Prerequisites
+
+- [ ] Ensure that your repo doesn't have too many:
+  - [ ] open PRs. The branch rename will re-trigger prow on _all_ PRs,
+    which will cause a huge spike in the CI load.
+  - [ ] periodic jobs. Changes to frequent periodics haven't been tested yet.
+
+- [ ] Create an issue in the [kubernetes/org] repo to track the branch
+rename for your repo.
+
+- [ ] Assign the GitHub Management team ([@kubernetes/owners]) for approval.
+
+- [ ] Once the issue has been approved, send a notice to your SIG's
+mailing list about the potential branch rename.
+
+## Changes pre-rename
+
+Make the following changes _before_ renaming the branch the `master` branch.
+
+Note: There might be additional changes required that have not been
+covered in this checklist.
+
+### Prowjobs
+
+Check if any of the prowjobs (presubmits, postsubmits or periodics)
+explicitly mention the `master` branch.
+
+- [ ] If a prowjob triggers on the `master` branch (`branches` field
+of the prowjob), add the `main` branch to the list
+(see [kubernetes/test-infra#20665] for an example).
+
+- [ ] If a prowjob mentions the `master` branch in `base_ref`,
+update it to the `main` branch. This change is disruptive so ensure
+that the branch is renamed between periodic job runs.
+
+- [ ] If a prowjob mentions `master` in its name, rename the job to
+include `main`. [`status-reconciler`] should automatically migrate the
+PR status contexts to the new job name but this has not been tested yet.
+This _may_ also impact testgrid history post-migration.
```

Comment:
@spiffxp do you know if this is really disruptive to testgrid?

Reply:
Testgrid is not aware of job renames / the results will be uploaded by CI to a location based on the name and appear as a different job to testgrid. Most repos probably don't have branch names in the jobs, but if they do they will effectively throw away history.

Reply:
We could exclude the branch name entirely from the default branch jobs, but that's somewhat inconsistent with jobs that include their (release) branches. I don't think we're likely to rename branches again in the future?

Reply:
Added text about testgrid behavior.